Expensify/expensify-common
# [feat] Add formatting to markdown link text

@parasharrajat  will you please review this?

This PR tries to add a feature to apply selected formatting rules in the markdown parser and uses this feature to format the link text.

`ExpensiMark` replace method now accepts an optional argument.

```
    /**
     * Replaces markdown with html elements
     *
     * @param {String} text - Text to parse as markdown
     * @param {Object} [options] - Options to customize the markdown parser
     * @param {String[]} [options.filterRules=[]] - An array of name of rules as defined in this class.
     * If not provided, all available rules will be applied.
     * @param {Boolean} [options.isTextEscaped=false] - Whether the text is already escaped
     *
     * @returns {String}
     */
```

### Fixed Issues
$ https://github.com/Expensify/App/issues/8426

### Related PR
$ https://github.com/Expensify/App/pull/9022

# Tests
1. Open the app
2. Select a report
3. Send `[~Hi~ **Hello** _Expensify_](expensify.com)`
4. Verify that `~Hi~ **Hello** _Expensify_` is formatted as per the rules.

# QA
1. Open the app
2. Select a report
3. Send `[~Hi~ **Hello** _Expensify_](expensify.com)`
4. Verify that `~Hi~ **Hello** _Expensify_` is formatted as per the rules.

# Screenshots
### Web
<img width="1440" alt="web" src="https://user-images.githubusercontent.com/77761491/168789656-7eec1005-b66c-47a7-965a-f306167e20bd.png">

### Desktop
![desktop](https://user-images.githubusercontent.com/77761491/168789655-850dd7fe-2ffa-43ec-83bc-5bb909d76dbe.jpg)

### Mobile web (Safari)
![mweb](https://user-images.githubusercontent.com/77761491/168789671-d4f5583c-4463-4423-a6bb-6466f824617a.png)

### iOS
![ios](https://user-images.githubusercontent.com/77761491/168789692-84ec7ab4-8fbe-46de-8157-c8955ff819f2.png)

### Android
![android](https://user-images.githubusercontent.com/77761491/168789705-d0974356-c180-422e-ac44-6ea097a66cab.png)
